backends/dpdk/dpdkArch.cpp

```diff
@@ -336,15 +336,64 @@ const IR::Node *ReplaceMetadataHeaderName::preorder(IR::Member *m) {
     return m;
 }
 
-const IR::Node *ReplaceMetadataHeaderName::preorder(IR::Parameter *p) {
-    if (auto type = p->type->to<IR::Type_Name>()) {
-        if (type->path->name == info->header_type) {
-            return new IR::Parameter(IR::ID("h"), p->direction, p->type);
-        } else if (type->path->name == info->local_metadata_type) {
-            return new IR::Parameter(IR::ID("m"), p->direction, p->type);
-        }
-    }
-    return p;
+/* This function replaces the header and metadata parameter names in control
+ * blocks with "h" and "m" respectively.
+ * It assumes that  ConvertToDpdkArch pass has converted the control blocks to
+ * this form for DPDK Architecture
+ *         control ingressDeparser(packet_out bufffer, header hdr, metadata md);
+ *         control egressDeparser(packet_out bufffer, header hdr, metadata md);
+ *         control ingress(header hdr, metadata md);
+ *         control egress(header hdr, metadata md);
+*/
+const IR::Node *ReplaceMetadataHeaderName::preorder(IR::Type_Control *c) {
+    auto applyParams = new IR::ParameterList();
+    auto param0 = c->applyParams->parameters.at(0);
+    cstring param0_type = param0->type->to<IR::Type_Name>()->path->name;
```

Comment:
this won't work if someone typedef-ed the type. It is more reliable to get the type from the typemap and then check that it is an extern type. But this will work for most cases.